py-polars/polars/dataframe/frame.py

```diff
@@ -10349,14 +10349,55 @@ def rows(
         else:
             return self._df.row_tuples()
 
+    @overload
+    def rows_by_key(
+        self,
+        key: ColumnNameOrSelector | Sequence[ColumnNameOrSelector],
+        *,
+        named: Literal[False] = ...,
+        include_key: bool = ...,
+        unique: Literal[False] = ...,
+    ) -> dict[Any, Iterable[tuple[Any, ...]]]: ...
```

Comment:
This is not correct, the values of the dictionary are not iterables of tuple, they should be a "list of Any":
```py
import polars as pl
df = pl.DataFrame({
       "w": ["a", "b"],
       "x": ["q", "q"]
})
print(repr(df.rows_by_key(key=["w"])))
# defaultdict(<class 'list'>, {'a': ['q'], 'b': ['q']})
```
Actually all return types here are not correct. See my previous comment on this PR for the correct types.

Reply:
arff, the type depends on the number of columns, see:

```python
import polars as pl
df = pl.DataFrame({
       "w": ["a", "b"],
       "x": ["q", "q"],
       "z": ["v", "v"]
})
print(repr(df.rows_by_key(key=["w"], named=False, unique=True)))
# {'a': ('q', 'v'), 'b': ('q', 'v')}
```
But in any case, you are right, this is not correct.